trustmanager/yubikey/yubikeystore.go

```diff
@@ -686,18 +711,32 @@ func (s *YubiKeyStore) addKey(keyID, role string, privKey data.PrivateKey) (
 	}
 	logrus.Debugf("Attempting to store key using yubikey slot %v", slot)
 
-	err = addECDSAKey(
-		ctx, session, privKey, slot, s.passRetriever, role)
-	if err == nil {
-		s.keys[privKey.ID()] = yubiSlot{
-			role:   role,
-			slotID: slot,
-		}
-		return true, nil
+	switch privKey.(type) {
+	case *data.ECDSAPrivateKey:
+		err = addECDSAKey(
+			ctx,
+			session,
+			privKey,
+			slot,
```

Comment:
I've been wondering if we should change the signatures for `addECDSAKey` and `createLoadCert` to have a `slotID` variable instead of `privKeyID` variable, WDYT?